arm/Microsoft.AAD/DomainServices/deploy.bicep

```
@@ -1,63 +1,112 @@
 @description('Optional. The name of the AADDS resource. Defaults to the domain name specific to the Azure ADDS service.')
 param name string = domainName
 
-@sys.description('Required. The domain name specific to the Azure ADDS service.')
+@description('Required. The domain name specific to the Azure ADDS service.')
 param domainName string
 
-@description('Required. The name of the sku specific to Azure ADDS Services - Standard is the default')
-param sku string
+@description('Optional. The name of the SKU specific to Azure ADDS Services.')
+@allowed([
+  'Standard'
+  'Enterprise'
+  'Premium'
+])
+param sku string = 'Standard'
 
-@description('Required. The location to deploy the Azure ADDS Services')
-param location string
+@description('Optional. The location to deploy the Azure ADDS Services')
+param location string = resourceGroup().location
 
 @description('Optional. Additional replica set for the managed domain')
 param replicaSets array = []
 
-@description('Required. The value is the base64encoded representation of the certificate pfx file')
-param pfxCertificate string
+@description('Conditional. The certificate required to configure Secure LDAP. Should be a base64encoded representation of the certificate PFX file. Required if secure LDAP is enabled and must be valid more than 30 days.')
```

Should we put the "Required if" right after the Conditional parameter type? That way it would also be possible to test it via Pester. What's your take on that?

```suggestion
@description('Conditional. Required if secure LDAP is enabled and must be valid more than 30 days. The certificate required to configure Secure LDAP. Should be a base64encoded representation of the certificate PFX file.')
```